### PR TITLE
[SDM] Add Android project for service discovery manager

### PR DESCRIPTION
**Pull request by younghajung**
This patch adds Anroid project for service discovery manager.
It can be opened to test using Android studio.
In order to test on non-rooted Android device, needs extra step.

On host PC:
\#adb shell touch /data/local/tmp/chrome-command-line

On target device:
Settings > Developer options > Select app to be debugged > Choose
Chromium

Signed-off-by: yh106.jung <yh106.jung@samsung.com>

### CODE REVIEW

File: service_discovery_manager/Component/mmDiscovery/service_server.cpp
Comment (Chikdolman) [inline service_discovery_manager/Component/mmDiscovery/service_server.cpp:101]: will the directory(data/local/tmp) mostly exist on Android? 
Would it be better to check and mkdir if dir doesn't exist? (Not sure if we have a permission to mkdir)

File: service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp
Comment (Chikdolman) [inline service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp]: kClassName instead of "app/samsung/org/servicediscovery/SDServerService"?

File: service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp
Comment (Chikdolman) [inline service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp]: kClassName instead of "app/samsung/org/servicediscovery/SDServerService"?

File: service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp
Comment (Chikdolman) [inline service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp:29]: kClassName never used

**Review by Chikdolman — COMMENTED** [review]
Please check first review comments.

File: service_discovery_manager/Component/mmDiscovery/service_server.cpp
Comment (younghajung) [reply service_discovery_manager/Component/mmDiscovery/service_server.cpp:101]: It is a default file path to initialize chromium commandline.
Please see base/android/java/src/org/chromium/base/CommandLineInitUtil.java.

Comment (Chikdolman) [reply service_discovery_manager/Component/mmDiscovery/service_server.cpp:101]: Checked 👍

File: service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp
Comment (younghajung) [reply service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp]: Done

File: service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp
Comment (younghajung) [reply service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp:29]: Done

File: service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp
Comment (younghajung) [reply service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp]: Done

**Review by djmixkim — COMMENTED** [review]
lgtm, but wait for an approval of @Chikdolman

**Review by Chikdolman — APPROVED** [review]
Looks good to me for now. How about to add "android.intent.action.BOOT_COMPLETED" filter in the next version?

**Comment by younghajung** [comment]
> Looks good to me for now. How about to add "android.intent.action.BOOT_COMPLETED" filter in the next version?

Thanks, it will be considered later.

**Pull request merged**
